docs/content/guides/developer/sui-101/simulating-refs.mdx

```diff
@@ -1,6 +1,126 @@
+    - Mutable reference (`&mut`): You can alter the object (according to the API) but you can't destroy or transfer it. 
+    - Immutable reference (`&`): Further restricts the set of operations and the guarantees/invariants over the referenced object. You have read-only access to the object's data.
+
+Programmable transaction blocks (PTBs) do not currently allow the use of object references returned from one of its transaction commands. You can use input objects to the PTB, objects created by the PTB (like `MakeMoveVec`), or returned from a transaction command by value as references in subsequent transaction commands. If a transaction command returns a reference, however, you can't use that reference in any call, significantly limiting certain common patterns in Move.
+
+## The borrow module
+
+The Sui framework includes a [borrow](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/docs/borrow.md) module that offers a solution to the reference problem. The module provides access to an object by value but builds a model that makes it impossible to destroy, transfer, or wrap the object retrieved. The borrow module exposes a `Referent` object that wraps another object (the object you want to reference). The module uses the [hot potato pattern](/concepts/sui-move-concepts/patterns/hot-potato.mdx) (via a `Borrow` instance) to allow retrieval of the wrapped object by value. Within the same PTB, the module then forces the object to be returned to the `Referent`. The `Borrow` instance guarantees that the object returned is the same that was retrieved.
+
+As an example, consider the following module stub that exposes an object (`Asset`) and a function (`use_asset`) to use that object.
+
+```rust
+module a_module {
+    struct Asset has key, store {
+        … // some data
+    }
+
+    public fun use_asset(asset: &Asset) {
+        …. // some code
+    }
+}
+```
+
+The function `use_asset` takes an immutable reference to the asset (`&Asset`), which is a common pattern in a transaction command. 
+
+Now consider another module that uses this asset.
+
+```rust
+module another_module {
+    struct AssetManager has key {
+        asset: Asset,
+    }
+
+    public fun get_asset(manager: &AssetManager): &Asset {
+        &manager.asset
+    }
+}
+```
+
+This module creates an object (`AssetManager`) that references the object (`Asset`) created in the previous module (`a_module`).
+
+You could then write a Move function that creates an object by reference and passes it to the `use_asset` function.  
+
+```rust
+fun do_something(manager: &AssetManager) {
+    let asset = another_module::get_asset(manager);
+    a_module::use_asset(asset);
+}
+```
+
+This function is not valid within a PTB, however, because PTBs do not support a reference returned by a function and passed to another function.
```

Comment:
I would say it more like:
the 2 functions in `do_something`, however, are not valid within a PTB because...
or something similar?